test/core/xds/xds_client_test.cc

```diff
@@ -2695,6 +2719,85 @@ TEST_F(XdsClientTest, FederationChannelFailureReportedToWatchers) {
   EXPECT_TRUE(stream2->Orphaned());
 }
 
+TEST_F(XdsClientTest, AdsReadWaitsForHandleRelease) {
+  InitXdsClient();
+  // Start watches for "foo1" and "foo2".
+  auto watcher1 = StartFooWatch("foo1");
```

Comment:
Please check the first request seen by the server after starting the first watcher but before starting the second watcher.  In other words, instead of starting both watches and then checking both requests, please start a watch, check the request, then start the second watch, then start the second request.